### PR TITLE
Bug Fixes

### PR DESCRIPTION
**Pull request by ArchBear-Vega**
- Fixes AirScooter bouncing up and down
- Fixes Flight cooldown error on first progress iteration.
- Fixes WaterSpout and WaterWave movement interactions
- Fixed Chat Branding throwing a null pointer whenever an invalid color is used
- Added AddonAbility and Author messages to /b h [ability] if they have a valid author and are an addon.

### CODE REVIEW

File: src/com/projectkorra/projectkorra/airbending/AirScooter.java [hunk 1]
```diff
@@ -113,13 +112,14 @@ public void progress() {
...
 		Vector velocity = player.getEyeLocation().getDirection().clone().normalize();
+		velocity.setY(0);
 		velocity = velocity.clone().normalize().multiply(speed);
 		/*
 		 * checks the players speed and ends the move if they are going too slow
 		 */
 		if (System.currentTimeMillis() > getStartTime() + interval) {
 			if (player.getVelocity().length() < speed * 0.3) {
-				remove();
+			remove();
```
Comment (OmniCypher-) [inline src/com/projectkorra/projectkorra/airbending/AirScooter.java]: This is formatted incorrectly. You removed a tab that needed to be there.

File: src/com/projectkorra/projectkorra/airbending/flight/AirFlight.java [hunk 2]
```diff
@@ -108,20 +109,24 @@ public void progress() {
 		if (!bPlayer.canBend(this)) {
 			remove();
 			return;
-		} else if (!player.isSneaking() && !isHovering && !firstProgressIteration) {
```
Comment (OmniCypher-) [inline src/com/projectkorra/projectkorra/airbending/flight/AirFlight.java]: The `else if` formatting here is incorrect as well. Please change it back to how it was.

File: src/com/projectkorra/projectkorra/command/HelpCommand.java [hunk 3]
```diff
@@ -140,21 +141,28 @@ else if (args.size() == 0) {
 			if (!ability.getInstructions().isEmpty()) {
 				sender.sendMessage(ChatColor.GOLD + usage + ability.getInstructions());
 			}
+			if (ability instanceof AddonAbility) {
+				AddonAbility abil = (AddonAbility) CoreAbility.getAbility(arg);
+				
+				sender.sendMessage(color + "* ADDON ABILITY *");
```
Comment (OmniCypher-) [inline src/com/projectkorra/projectkorra/command/HelpCommand.java]: This should be a configurable option in the language file. We will need to talk about this in more detail in #devs.

File: src/com/projectkorra/projectkorra/command/HelpCommand.java [hunk 4]
```diff
@@ -140,21 +141,28 @@ else if (args.size() == 0) {
...
+				sender.sendMessage(ChatColor.GRAY + "- By " + abil.getAuthor());
+			}
+			
 		} else if (Arrays.asList(Commands.airaliases).contains(args.get(0))) {
 			sender.sendMessage(Element.AIR.getColor() + air.replace("/b help AirCombos", Element.AIR.getSubColor() + "/b help AirCombos" + Element.AIR.getColor()));
-			sender.sendMessage(ChatColor.YELLOW + learnMore + ChatColor.DARK_AQUA + "http://tinyurl.com/qffg9m3");
```
Comment (OmniCypher-) [inline src/com/projectkorra/projectkorra/command/HelpCommand.java]: These changes essentially make the message `Learn more: ` with no link to anything. Either the link needs to be fixed or the message needs to be changed.

File: src/com/projectkorra/projectkorra/waterbending/WaterSpoutWave.java [hunk 5]
```diff
@@ -131,7 +131,12 @@ public void progress() {
 			remove();
 			return;
 		}
-
+		if (CoreAbility.hasAbility(player, WaterSpout.class)) {
+		    WaterSpout waterSpout = CoreAbility.getAbility(player, WaterSpout.class);
+		    waterSpout.remove();
+		}
+		
+
```
Comment (OmniCypher-) [inline src/com/projectkorra/projectkorra/waterbending/WaterSpoutWave.java]: Need to remove one of these spaces.

File: src/com/projectkorra/projectkorra/airbending/flight/AirFlight.java [hunk 6]
```diff
@@ -30,6 +30,7 @@
 	private double hoverY;
 	@Attribute(Attribute.DURATION)
 	private double maxDuration;
+
```
Comment (OmniCypher-) [inline src/com/projectkorra/projectkorra/airbending/flight/AirFlight.java]: Need to remove one of these spaces.

File: src/com/projectkorra/projectkorra/GeneralMethods.java [hunk 7]
```diff
@@ -2030,8 +2030,13 @@ public static FallingBlock spawnFallingBlock(Location loc, Material type, byte d
 	}
 
 	public static void sendBrandingMessage(CommandSender sender, String message) {
-		ChatColor color = ChatColor.valueOf(ConfigManager.languageConfig.get().getString("Chat.Branding.Color").toUpperCase());
-		color = color == null ? ChatColor.GOLD : color;
+		ChatColor color;
```
Comment (OmniCypher-) [inline src/com/projectkorra/projectkorra/GeneralMethods.java]: Needs a space after the try and catch statement to follow our formatting standard of `try {` and `catch (Exception e) {`.

File: src/com/projectkorra/projectkorra/airbending/AirScooter.java [hunk 8]
```diff
@@ -153,7 +151,7 @@ public void progress() {
 			return;
 		}
 
-		player.setSprinting(false);
+
```
Comment (OmniCypher-) [inline src/com/projectkorra/projectkorra/airbending/AirScooter.java]: Need to remove one of these spaces.

File: src/com/projectkorra/projectkorra/command/HelpCommand.java (hunk 4 above)
Comment (ArchBear-Vega) [inline src/com/projectkorra/projectkorra/command/HelpCommand.java]: I didn't notice that sorry, Ill fix that XD

File: src/com/projectkorra/projectkorra/command/HelpCommand.java (hunk 3 above)
Comment (ArchBear-Vega) [inline src/com/projectkorra/projectkorra/command/HelpCommand.java]: Okay

**Pull request closed without merging**